homeassistant/components/delijn/sensor.py

```diff
@@ -83,20 +83,23 @@ async def async_update(self):
         self._attributes["stopname"] = self._name
 
         try:
-            first = self.line.passages[0]
-            if first["due_at_realtime"] is not None:
-                first_passage = first["due_at_realtime"]
+            if len(self.line.passages) > 1: """do a check if there is a passage"""
+                first = self.line.passages[0]
+                if first["due_at_realtime"] is not None:
+                    first_passage = first["due_at_realtime"]
+                else:
+                    first_passage = first["due_at_schedule"]
+                self._state = first_passage
+                self._attributes["line_number_public"] = first["line_number_public"]
+                self._attributes["line_transport_type"] = first["line_transport_type"]
+                self._attributes["final_destination"] = first["final_destination"]
+                self._attributes["due_at_schedule"] = first["due_at_schedule"]
+                self._attributes["due_at_realtime"] = first["due_at_realtime"]
+                self._attributes["is_realtime"] = first["is_realtime"]
+                self._attributes["next_passages"] = self.line.passages
+                self._available = True
             else:
-                first_passage = first["due_at_schedule"]
-            self._state = first_passage
-            self._attributes["line_number_public"] = first["line_number_public"]
-            self._attributes["line_transport_type"] = first["line_transport_type"]
-            self._attributes["final_destination"] = first["final_destination"]
-            self._attributes["due_at_schedule"] = first["due_at_schedule"]
-            self._attributes["due_at_realtime"] = first["due_at_realtime"]
-            self._attributes["is_realtime"] = first["is_realtime"]
-            self._attributes["next_passages"] = self.line.passages
-            self._available = True
+                self._available = False
         except (KeyError, IndexError):
```

Comment:
If we have a check for list members we don't need to catch IndexError here.

Reply:
That seems be a correct statement (and best practice, not to throw exceptions if not needed), but that I will leave to the code owners. Don't know if this try and catch code catches other things....